Assignment 3.Rmd

```diff
@@ -127,29 +131,94 @@ Now we can plot our line plot using the ggplot command, "ggplot()".
+```
+
+Look at the number of people in each cluster, now repeat this process for 3 rather than 2 clusters. 
+
+```{r}
+fit <- kmeans(K3, 3)
+fit$cluster
+K8 <- data.frame(K3, fit$cluster)
+names(K8) <- c("1", "2", "3", "4", "5", "cluster")
+K9 <- K8 %>% gather(Week,"Motivation_Score",1:5)
+K10 <- K9 %>% group_by(Week, cluster) %>% mutate(Average = mean(Motivation_Score))
+K10$Week <- as.numeric(K10$Week)
+K10$cluster <- as.factor(K10$cluster)
+ggplot(K10, aes(x=Week, y=Average, color=cluster)) + 
+  geom_line() + labs(x="Week", y="Average Motivation")
+K11 <- K10 %>% group_by(cluster) %>% count(cluster)
 ```
 
-Look at the number of people in each cluster, now repeat this process for 3 rather than 2 clusters. Which cluster grouping do you think is more informative? Write your answer below:
+Which cluster grouping do you think is more informative? Write your answer below:
+
+I believe the 3-cluster grouping is more informative. Because we can see there's a group of people has an obvious decrease from week 3 to week 5. But if we use the 2-cluster one, we won't see this.
+
 
 ##Part II
 
 Using the data collected for Assignment 2 (which classes students were in), cluster the students, then redraw the graph of the class but color the students according the cluster they are in.
 
+```{r}
+L1 <- read.csv("hudk4050-classes.csv", header=TRUE)
+L1 <- L1[-c(34),]
+L2 <- L1%>%gather(class_number,class_title,Class.1:Class.6,-First.Name,-Last.Name) %>%
+   filter(class_title !="") %>%
+  dplyr::select(First.Name, class_title)
+L2$class_title <- gsub(' ', '', L2$class_title)
+L2$class_title <- gsub('&', '', L2$class_title)
+L2 <- L2 %>% filter(class_title != "HUDK4050")
+L3 <- L2 %>% mutate(count = 1) %>%
+  spread(class_title, count)
+rownames(L3) <- L3$First.Name
+L3 <- data.matrix(L3[,-1])
+L3[is.na(L3)] <- 0
+L4 <- L3 %*% t(L3)
+diag(L4) <- 0
+fit <- kmodes(L4,5,iter.max=10,weighted=FALSE)
+g <- graph.adjacency(L4, mode="undirected")
+plot(g, vertex.color = fit$cluster, vertex.label.cex = 0.8)
+
+```
 ##Part III
 
 In class activity 6 you clustered students in the class by the answers to a questionaire. Create a visualization that shows the overlap between these clusters and the clusters generated in part II.
 
+```{r}
+class_cluster = fit$cluster
+names(class_cluster) <- rownames(L4)
+M1 <- read.csv("HUDK405019-clustering.csv", header = TRUE)
+M1 <- unite(M1, "Name", c("First.Name", "Last.Name"), sep = ".")
+row.names(M1) <- M1$Name
+M1$Name <- NULL
+M2 <- dplyr::select(M1, 1:11) %>% 
+  mutate_all(funs(gsub("[a-zA-Z]", "", .))) %>% 
+  mutate_all(funs(as.numeric(.)))
+M2 <- as.data.frame(scale(M2))
+M2 <- M2 %>% 
+  mutate_all(funs(ifelse(is.na(.) == TRUE, 0, .)))
+class_cluster <- class_cluster[1:50]
+fit2 <- kmeans(M2, 5) 
+fit2$cluster
+location_cluster = fit2$cluster
+names(location_cluster) <- rownames(L4[1:50])
+struc <- structable(class_cluster ~ location_cluster)
+mosaic(struc, shade=TRUE, legend=TRUE)
+
+```
```

Comment:
Overall. great job on the coding part of this assignment but you may want to work on the analysis aspect of it. Nonetheless, keep up the good work!